Framework/src/Framework/HistogramPool.cxx

```diff
@@ -0,0 +1,299 @@
+
+#include "Framework/HistogramPool.h"
+
+//----------------//
+//   C++ StdLib   //
+//----------------//
+#include <stdexcept>
+
+//----------//
+//   ROOT   //
+//----------//
+#include "TH1.h"
+#include "TStyle.h"
+
+namespace framework {
+
+TH1* HistogramPool::get(const std::string& name) {
+  auto histo = histograms_.find(name);
+  if (histo == histograms_.end()) {
+    EXCEPTION_RAISE(
+        "InvalidArg",
+        "Histogram " + name +
+            " not found in pool."
+            "\nMake sure to `histograms_.create` in onProcessStart for any "
+            "histogram you want to `histograms_.fill`.");
+  }
+
+  return histograms_[name];
+}
+
+void HistogramPool::insert(const std::string& name,
+                           std::function<TH1*()> factory, bool weighted) {
+  if (histograms_.find(name) != histograms_.end()) {
+    EXCEPTION_RAISE(
+        "RepeatName",
+        "Histogram " + name +
+            " already exists in histogram pool."
+            "\nMake sure to use distinct names for your different histograms!"
+            "\nYou can use `histograms_.get` to retrieve a pointer to a "
+            "specific histogram "
+            "if you want to do some other customizations besides filling.");
+  }
+
+  get_directory_()->cd();
+
+  auto h = factory();
```

Comment:
can we have this as `histo` or something to avoid the one-letter variable names?